jetty-core/jetty-fcgi/jetty-fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/internal/ServerFCGIConnection.java

```diff
@@ -214,17 +221,22 @@ void parseAndFill()
     {
         if (LOG.isDebugEnabled())
             LOG.debug("parseAndFill {}", this);
+        acquireInputBuffer();
```

Comment:
Why is the input buffer not release when parse below returns true?

Reply:
You're right; the buffer could be released when empty and the parser returns true.